lib/src/core/authentication-core.ts

```diff
@@ -197,6 +200,12 @@ export class AuthenticationCore<T> {
         body.set("grant_type", "authorization_code");
         body.set("redirect_uri", configData.signInRedirectURL);
 
+        if (tokenRequestConfig && tokenRequestConfig.params) {
```

Comment:
Maybe we can go for a shorthand here. WDYT?

```suggestion
        if (tokenRequestConfig?.params) {
```